File: web/src/main/angular/app/model/query-report.ts
import {getFhirNow} from '../helper';

export class QueryReport {
  public date: string = getFhirNow();
  public measureId: string;
  public questions: string[] = [];
  public answers: { [questionId: string]: any } = {};

  constructor(... questionIds: string[]) {
    this.questions = questionIds || [];
  }

  hasAnswer(questionId: string): boolean {
    return this.answers.hasOwnProperty(questionId) && this.answers[questionId] !== null;
  }
}

File: core/src/main/java/com/lantanagroup/link/QueryReport.java
package com.lantanagroup.link;

import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Getter
@Setter
public class QueryReport {
  private String date;
  private String measureId;
  private List<String> questions = new ArrayList<>();
  private Map<String, Object> answers = new HashMap<>();

  public void setAnswer(String questionId, Object value) {
    this.answers.put(questionId, value);
  }

  public Object getAnswer(String questionId) {
    return this.answers.get(questionId);
  }
}
